File: LeetCode/cpp/lc_0026.cpp
/**
 * @file     lc_0026.cpp
 * @brief    https://leetcode-cn.com/problems/remove-duplicates-from-sorted-array/
 * @author   YongDu
 * @date     2021-09-05
 */
class Solution {
public:
  int removeDuplicates(vector<int> &nums) {
    if (nums.empty()) {
      return 0;
    }
    int index = 0;
    for (int i = 1; i < nums.size(); ++i) {
      if (nums[i] != nums[index]) {
        nums[++index] = nums[i];
      }
    }
    return index + 1;
  }
};

File: LeetCode/python/lc_0026.py
"""
 * @file     lc_0026.py
 * @brief    
 * @author   YongDu
 * @date     2021-10-29
"""


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        index = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[index]:
                index += 1
                nums[index] = nums[i]
        return index + 1
